### PR TITLE
Add support to run benchmarks using OpenSearch distribution url

### LINKED ISSUE #4073: Add option to run benchmark against OS artifact url
The nightly benchmarks currently depend on the distribution manifest file to get the location of the OpenSearch distribution tarball and then set up the cluster and run benchmark. 
There have been a few requests to run ad-hoc benchmark tests using nightly benchmark framework against the distribution tarball provided by requester. Since the tarball was created locally and had no manifest associated with it we could not accept the request and guided the user to run it locally. 
Add a feature in the workflow to be able to run the benchmark workflow using either a distribution manifest url or the actual artifact url.

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
In the current phase the nightly benchmark runs are tightly coupled with distribution manifest file generated as part of our OpenSearch distribution build process. 
There have been a request from developers to be able to use our benchmark platform to run ad-hoc benchmarks to test their local changes. For this the most basic requirement is to directly pass the OpenSearch distribution url instead of code reading it from `manifest.yml` file. 
The existing functionality would still work as the user can either use distribution manifest or distribution url as inputs. 

We are defaulting to `x64` arch if distribution url is provided as input and will add support for `arm64` in subsequent PRs. 

### Issues Resolved
#4073 
https://github.com/opensearch-project/OpenSearch/issues/9418

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test_workflow/benchmark_test/benchmark_args.py [hunk 1]
```diff
@@ -131,3 +137,8 @@ def __init__(self) -> None:
         self.telemetry = args.telemetry
         self.telemetry_params = args.telemetry_params if args.telemetry_params else None
         self.logging_level = args.logging_level
+
+        if self.bundle_manifest is None and self.distribution_url is None:
+            raise Exception('Please provide either --bundle-manifest or --distribution-url to run the performance test.')
```
Comment (peterzhuamazon) [inline src/test_workflow/benchmark_test/benchmark_args.py:142]: What if both are not NONE, then which one will be taken?

File: src/test_workflow/benchmark_test/benchmark_test_cluster.py [hunk 2]
```diff
@@ -112,27 +114,37 @@ def wait_for_processing(self, tries: int = 3, delay: int = 15, backoff: int = 2)
         logging.info(f"Waiting for domain at {self.endpoint} to be up")
         protocol = "http://" if self.args.insecure else "https://"
         url = "".join([protocol, self.endpoint, "/_cluster/health"])
-        request_args = {"url": url} if self.args.insecure else {"url": url, "auth": HTTPBasicAuth("admin", "admin"), "verify": False}  # type: ignore
+        request_args = {"url": url} if self.args.insecure else {"url": url, "auth": HTTPBasicAuth("admin", "admin"),  # type: ignore
```
Comment (peterzhuamazon) [inline src/test_workflow/benchmark_test/benchmark_test_cluster.py:117]: Verify:False not needed?

File: src/test_workflow/benchmark_test/benchmark_args.py (hunk 1 above)
Comment (rishabh6788) [reply src/test_workflow/benchmark_test/benchmark_args.py:142]: If you are talking about `--bundle-manifest` and `--distribution-version` being provided at the same time then workflow will default to `--bundle-manifest` and use that to run.

File: src/test_workflow/benchmark_test/benchmark_test_cluster.py (hunk 2 above)
Comment (rishabh6788) [reply src/test_workflow/benchmark_test/benchmark_test_cluster.py:117]: It is there on line #118

File: src/test_workflow/benchmark_test/benchmark_test_cluster.py [hunk 3]
```diff
@@ -112,27 +114,37 @@ def wait_for_processing(self, tries: int = 3, delay: int = 15, backoff: int = 2)
...
-        else:
+        elif self.manifest:
             suffix = self.manifest.build.id + '-' + self.manifest.build.architecture
+        elif self.args.stack_suffix:
+            suffix = self.args.stack_suffix
+
+        if self.manifest:
+            artifact_url = self.manifest.build.location if isinstance(self.manifest, BundleManifest) else \
+                f"https://artifacts.opensearch.org/snapshots/core/opensearch/{self.manifest.build.version}/opensearch-min-" \
+                f"{self.manifest.build.version}-linux-{self.manifest.build.architecture}-latest.tar.gz"
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply src/test_workflow/benchmark_test/benchmark_test_cluster.py:134]: Unfortunately the build manifest doesn't have `location` attribute like bundle-manifest, so had to fall back to using formatted url.

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**